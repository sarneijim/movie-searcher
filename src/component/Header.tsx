import React, { Component } from "react";
import {
  AppBar,
  Toolbar,
  Typography,
  InputBase,
  IconButton,
  FormControlLabel,
  RadioGroup,
  Radio,
} from "@material-ui/core";
import SearchIcon from "@material-ui/icons/Search";
import { updateCatalog } from "../store/actionCreators";
import { Link } from "react-router-dom";
import styles from "../styles/header.module.css";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

class Header extends Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = {
      filter: {
        text: "",
        type: "movie",
      },
    };
  }
  render() {
    const inputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      this.setState({
        filter: {
          ...this.state.filter,
          text: event.target.value,
        },
      });
      // TODO AC11 & AC12: When filter.length>4 add autocomplete input
    };
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      this.setState(
        {
          filter: {
            ...this.state.filter,
            type: event.target.value,
          },
        },
        () => handleSubmit(event)
      );
    };
    const handleSubmit = (event: any = {}) => {
      event.preventDefault();
      this.props.history.push("/");
      this.props.updateCatalog(this.state.filter);
    };
    return (
      <AppBar position="static">
        <Toolbar>
          <Link to={`/`}>
            <Typography className={styles.title} variant="h6">
              Movies
            </Typography>
          </Link>
          <form
            className={styles.form}
            autoComplete="off"
            onSubmit={handleSubmit}
          >
            <div className={styles.search}>
              <InputBase
                placeholder="Searchâ€¦"
                autoFocus
                inputProps={{ "aria-label": "search" }}
                onChange={inputChange}
              />
              <IconButton
                aria-label="search"
                color="inherit"
                onClick={handleSubmit}
              >
                <SearchIcon />
              </IconButton>
            </div>

            <RadioGroup
              aria-label="type"
              name="type"
              value={this.state.filter.type}
              onChange={handleChange}
              className={styles.group}
            >
              <FormControlLabel
                value="movie"
                control={<Radio />}
                label="Movie"
              />
              <FormControlLabel value="show" control={<Radio />} label="Show" />
              <FormControlLabel
                value="actor"
                control={<Radio />}
                label="Actor"
              />
              <FormControlLabel value="all" control={<Radio />} label="All" />
            </RadioGroup>
          </form>
        </Toolbar>
      </AppBar>
    );
  }
}
const mapDispatchToProps = (dispatch: any) => {
  return {
    updateCatalog: (filter: any) => dispatch(updateCatalog(filter)),
  };
};

export default withRouter(connect(null, mapDispatchToProps)(Header));
