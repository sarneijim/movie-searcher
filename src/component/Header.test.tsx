import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import Header from "./Header";
import { BrowserRouter, Route } from "react-router-dom";
const mockStore = configureStore([]);

describe("Header", () => {
  let component: any;
  const store = mockStore({});

  beforeEach(() => {
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>
    );
  });

  afterEach(() => {
    component = null;
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match the snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
