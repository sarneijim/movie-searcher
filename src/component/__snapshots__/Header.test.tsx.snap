// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h6
        className="MuiTypography-root title MuiTypography-h6"
      >
        Movies
      </h6>
    </a>
    <form
      autoComplete="off"
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="search"
      >
        <div
          className="MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            autoFocus={true}
            className="MuiInputBase-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Searchâ€¦"
            type="text"
          />
        </div>
        <button
          aria-label="search"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        aria-label="type"
        className="MuiFormGroup-root group"
        role="radiogroup"
      >
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input-4"
                name="type"
                onChange={[Function]}
                type="radio"
                value="movie"
              />
              <div
                className="PrivateRadioButtonIcon-root-5 PrivateRadioButtonIcon-checked-7"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Movie
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-4"
                name="type"
                onChange={[Function]}
                type="radio"
                value="show"
              />
              <div
                className="PrivateRadioButtonIcon-root-5"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Show
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-4"
                name="type"
                onChange={[Function]}
                type="radio"
                value="actor"
              />
              <div
                className="PrivateRadioButtonIcon-root-5"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Actor
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-4"
                name="type"
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <div
                className="PrivateRadioButtonIcon-root-5"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            All
          </span>
        </label>
      </div>
    </form>
  </div>
</header>
`;
